community/values/src/main/java/org/neo4j/values/storable/DateTimeValue.java

```diff
@@ -493,11 +492,19 @@ int unsafeCompareTo( Value other )
         return cmp;
     }
 
-    private boolean areDifferentZones( ZoneId thisZone, ZoneId thatZone )
+    private boolean areSameZone( ZoneId thisZone, ZoneId thatZone )
     {
-        return !(thisZone instanceof ZoneOffset) &&
-               !(thatZone instanceof ZoneOffset) &&
-                TimeZones.map( thisZone.getId() ) != TimeZones.map( thatZone.getId() );
+        boolean thisIsOffset = thisZone instanceof ZoneOffset;
+        boolean thatIsOffset = thatZone instanceof ZoneOffset;
```

Comment:
`thisIsOffset` and `thatIsOffset` are already evaluated in `unsafeCompareTo`. Send them in as two boolean parameters to this function instead of evaluating them agan